trunk/SUAVE/Input_Output/VTK/save_prop_vtk.py

```diff
@@ -0,0 +1,264 @@
+       filename      Name of vtk file to save                           [String]  
+       Results       Data structure of wing and propeller results       [Unitless]
+       i_prop        ith propeller to evaluate wake of                  [Unitless]
+       time_step     Simulation time step                               [Unitless]
+       
+    Outputs:                                   
+       N/A
+
+    Properties Used:
+       N/A 
+    
+    Assumptions:
+       Quad cell structures for mesh
+    
+    Source:  
+       None
+    
+    """  
+    
+    # Generate propeller point geometry
+    n_blades = prop.number_of_blades 
+    Gprops   = generate_lofted_propeller_points(prop)
+    
+    for B_idx in range(n_blades):
+        # Get geometry of blade for current propeller instance
+        G = Gprops[i_prop][B_idx]
+        
+        sep  = filename.find('.')
+        file = filename[0:sep]+"_blade"+str(B_idx)+"_t"+str(time_step)+filename[sep:]
+        
+        # Create file for each blade
+        with open(file, 'w') as f:
+            #---------------------
+            # Write header
+            #--------------------- 
+            header = ["# vtk DataFile Version 4.0"                ,   # File version and identifier
+                      "\nSUAVE Model of PROWIM Propeller Blade "  ,   # Title 
+                      "\nASCII"                                   ,   # Data type
+                      "\nDATASET UNSTRUCTURED_GRID"               ]   # Dataset structure / topology     
+
+            f.writelines(header)     
+            
+            # --------------------
+            # Write Points
+            # --------------------   
+            n_r      = len(prop.chord_distribution)
+            n_af     = Gprops.n_af
+            
+            n_vertices    = (n_r)*(n_af)    # total number of node vertices per blade
+            points_header = "\n\nPOINTS "+str(n_vertices) +" float"
+            f.write(points_header)
+            
+            # Loop over all nodes
+            for c_idx in range(n_af):
+                for r_idx in range(n_r):
+                    xp = round(G.X[r_idx,c_idx],4)
+                    yp = round(G.Y[r_idx,c_idx],4)
+                    zp = round(G.Z[r_idx,c_idx],4)
+                        
+                    new_point = "\n"+str(xp)+" "+str(yp)+" "+str(zp)
+                    f.write(new_point)  
+                        
+            #---------------------    
+            # Write Cells:
+            #---------------------
+            cells_per_blade = n_af*(n_r-1)
+            v_per_cell      = 4    # quad cells
+            size            = cells_per_blade*(1+v_per_cell) # total number of integer values required to represent the list
+            cell_header     = "\n\nCELLS "+str(cells_per_blade)+" "+str(size)
+            f.write(cell_header)
+            
+            rlap=0
+            for i in range(cells_per_blade): # loop over all nodes in blade
```

Comment:
This loop and the one in the script above are quite similar, can you make a single function that handles both scripts?

Reply:
I broke this into new function and called into prop and fuselage scripts